sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryContext.cs

```diff
@@ -1,14 +1,82 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.ClientModel.Primitives;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 
 namespace Azure.Core.Expressions.DataFactory
 {
     /// <summary>
     /// Context class used by <see cref="ModelReaderWriter"/> to read and write models in an AOT compatible way.
     /// </summary>
-    public partial class DataFactoryContext : ModelReaderWriterContext
+    public class DataFactoryContext : ModelReaderWriterContext
     {
+        private readonly Dictionary<Type, Func<ModelReaderWriterTypeBuilder>> _typeBuilderFactories = new();
+        private readonly ConcurrentDictionary<Type, ModelReaderWriterTypeBuilder> _typeBuilders = new();
+
+        private static readonly Dictionary<Type, ModelReaderWriterContext> s_referenceContexts = new()
+        {
+            { typeof(AzureCoreContext), AzureCoreContext.Default },
+        };
+
+        private static DataFactoryContext? _dataFactoryContext;
+        /// <summary> Gets the default instance </summary>
+        public static DataFactoryContext Default => _dataFactoryContext ??= new();
+
+        private DataFactoryContext()
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override bool TryGetTypeBuilderCore(Type type, out ModelReaderWriterTypeBuilder? builder)
```

Comment:
My intention is to add custom logic in this method, the only way I can think of now is to disable the generation for context.
@m-nash Please share your thoughts on this.